### PR TITLE
nvgpu backend check if the ptr pass to create_tensor is device pointer

### PR DESCRIPTION
**Pull request by fengleitian**

### CODE REVIEW

File: src/ngraph/runtime/gpu/gpu_backend.cpp [hunk 1]
```diff
@@ -114,6 +115,10 @@ shared_ptr<runtime::Tensor>
 shared_ptr<runtime::Tensor> runtime::gpu::GPU_Backend::create_tensor(
     const element::Type& element_type, const Shape& shape, void* memory_pointer)
 {
+    if (memory_pointer != nullptr && !is_device_pointer(memory_pointer))
+    {
+        throw ngraph_error("The pointer passed to create_tensor is not a device pointer.");
```
Comment (rkimballn1) [inline src/ngraph/runtime/gpu/gpu_backend.cpp]: We can create a device tensor here and copy the data over, but that might be a separate PR.

File: src/ngraph/runtime/gpu/gpu_tensor.cpp [hunk 2]
```diff
@@ -39,11 +39,15 @@ runtime::gpu::GPUTensor::GPUTensor(const ngraph::element::Type& element_type,
         std::make_shared<ngraph::descriptor::layout::DenseTensorLayout>(*m_descriptor));
 
     m_buffer_size = shape_size(shape) * element_type.size();
-    if (memory_pointer != nullptr)
+    if (memory_pointer != nullptr && is_device_pointer(memory_pointer))
```
Comment (rkimballn1) [inline src/ngraph/runtime/gpu/gpu_tensor.cpp]: Have one if with memory_pointer != nullptr then inside check for is_device_pointer. It will look a little cleaner.

File: src/ngraph/runtime/gpu/gpu_util.cpp [hunk 3]
```diff
@@ -62,6 +62,17 @@ void runtime::gpu::free_gpu_buffer(void* buffer)
     }
 }
 
+bool runtime::gpu::is_device_pointer(const void *ptr)
+{
+  cudaPointerAttributes attributes;
```
Comment (rkimballn1) [inline src/ngraph/runtime/gpu/gpu_util.cpp]: Indenting looks wrong.

File: src/ngraph/runtime/gpu/gpu_util.cpp [hunk 4]
```diff
@@ -62,6 +62,17 @@ void runtime::gpu::free_gpu_buffer(void* buffer)
...
+  CUDA_RT_SAFE_CALL_NO_THROW(cudaPointerGetAttributes(&attributes, ptr));
+  if(attributes.devicePointer != NULL)
```
Comment (rkimballn1) [inline src/ngraph/runtime/gpu/gpu_util.cpp]: nullptr, not NULL

Comment (fengleitian) [reply src/ngraph/runtime/gpu/gpu_util.cpp]: sharp eye. missed "style-apply"

Comment (fengleitian) [reply src/ngraph/runtime/gpu/gpu_util.cpp]: fixed.

File: src/ngraph/runtime/gpu/gpu_util.cpp (hunk 3 above)
Comment (fengleitian) [reply src/ngraph/runtime/gpu/gpu_util.cpp]: sharp eye. missed "style-apply"

File: src/ngraph/runtime/gpu/gpu_tensor.cpp (hunk 2 above)
Comment (fengleitian) [reply src/ngraph/runtime/gpu/gpu_tensor.cpp]: agree. fixed.

File: src/ngraph/runtime/gpu/gpu_backend.cpp (hunk 1 above)
Comment (fengleitian) [reply src/ngraph/runtime/gpu/gpu_backend.cpp]: ya, I was also think in that way. one concern was user might assume the pointer will be used by backend. which is not true. it's likely user will miss the data sync.

Comment (fengleitian) [reply src/ngraph/runtime/gpu/gpu_backend.cpp]: We can handle that in another PR.

**Comment by AyanmoI** [comment]
LGTM
